Configuration File tracking upon Pull Requests

The-Huginn opened the pull request:
Resolves #54 

Also it seems it would be possible to create RestClient and retrieve raw URL such as [this](https://raw.githubusercontent.com/The-Huginn/wildfly-github-bot/b167e4aa848bcdd4b3dacdd6edb4032bf045b9df/.github/wildfly-bot.yml) to get the file directly using `GHPullRequestFileDetail#getRawUrl`.

I realized this after the implementation with patch was done. However I don't think it would work for private repos, as the `GitHub` objects contain authorization tokens. But it might be possible to retrieve this token to use it for this RestClient.

xstefank commented on src/main/java/io/xstefank/wildfly/bot/ConfigFileChangeProcessor.java:
```diff
@@ -0,0 +1,103 @@
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+@Default
```
Do you need default if you don't use alternatives?

The-Huginn replied:
Sorry, trailing from previous testing. I will remove it.

xstefank commented on src/main/java/io/xstefank/wildfly/bot/ConfigFileChangeProcessor.java:
```diff
@@ -0,0 +1,103 @@
+                try {
+                    String updatedFile = patch.applyTo(currentFileLines)
+                            .stream()
+                            .collect(Collectors.joining("\n"));
```
You can't get the updated file from diff? From the PR branch or something? This seems a little bit unnecessary.

The-Huginn replied:
I have created an issue for that, you can see here https://github.com/quarkiverse/quarkus-github-app/issues/486

xstefank commented on src/main/java/io/xstefank/wildfly/bot/ConfigFileChangeProcessor.java:
```diff
@@ -0,0 +1,103 @@
+                            .stream()
+                            .collect(Collectors.joining("\n"));
+                    Optional<WildFlyConfigFile> file = Optional.ofNullable(yamlObjectMapper.readValue(updatedFile, WildFlyConfigFile.class));
+                    List<String> invalidRules = validateFile(file.get());
```
If the file already parses it should be validated. Why do we not marking ids as required?

The-Huginn replied:
Now we have a few fields, which we can validate using `hibernate-validator`. id, email or description. I think we should create a separate issue for that.

xstefank commented on src/main/java/io/xstefank/wildfly/bot/ConfigFileChangeProcessor.java:
```diff
@@ -0,0 +1,103 @@
+        return result;
+    }
+
+    private void updateFormatCommitStatus(GHPullRequest pullRequest, GHCommitState commitState, String description) throws IOException {
```
we should probably create some utility that we can reuse in all processors

The-Huginn replied:
I will do it in this PR then.

xstefank commented on src/test/java/io/xstefank/wildfly/bot/PRDescriptionCheckTest.java:
```diff
@@ -75,7 +89,12 @@ void correctLinkCheckSuccessTest() throws IOException {
+                    mocks.configFileFromString("wildfly-bot.yml", wildflyConfigFile);
+
+                    PagedSearchIterable<GHPullRequestFileDetail> fileDetails = GitHubAppMockito.mockPagedIterable(mockEmptyFileDetails());
+                    Mockito.when(mocks.pullRequest(1352150111).listFiles()).thenReturn(fileDetails);
```
why do you need this? do we pass json that includes changes to the config file?

The-Huginn replied:
As otherwise all the tests would fail on following line in `ConfigFileChangeProcessor` as `listFiles` would returns by default null.
```suggestion
                    for (GHPullRequestFileDetail changedFile : pullRequest.listFiles()) {
```

xstefank commented on src/main/java/io/xstefank/wildfly/bot/config/util/RuntimeProcessor.java:
```diff
@@ -0,0 +1,17 @@
+import java.io.IOException;
+
+@ApplicationScoped
+public class RuntimeProcessor {
```
Maybe a better less generic name? This is GH utility class IIUC

xstefank commented:
@The-Huginn always don't forget to squash commits

The-Huginn commented:
I have removed the diff applier for patches and followed suggestion from [this issue](https://github.com/quarkiverse/quarkus-github-app/issues/486). Following that I also updated tests

xstefank commented on src/test/java/io/xstefank/wildfly/bot/PRConfigFileChangeTest.java:
```diff
@@ -0,0 +1,139 @@
+                    "      message: \"Too many commits in PR!\"\n" +
+                    "  emails:\n" +
+                    "    - rastislavbudinsky@gmail.com\n" +
+                    "    - rbudinsk@redhat.com",
```
you probably don't want real emails in tests

The-Huginn replied:
Thanks for noticing.
